app/ts/components/TransactionPage/TransactionDetails.tsx

```diff
@@ -0,0 +1,129 @@
+		case 'rejected':
+			return <InfoError displayText='Failed to load information' message={response.value.error.message} />
+		case 'resolved':
+			return (
+				<>
+					<Info label='Hash' value={response.value.value.hash} allowCopy />
+					<Info label='From' value={response.value.value.from} allowCopy />
+					<AccountBalance response={response.value.value} />
+				</>
+			)
+	}
+}
+
+const DataFromReceipt = ({ receipt }: { receipt: Signal<AsyncProperty<TransactionReceipt>> }) => {
+	switch (receipt.value.state) {
+		case 'inactive':
+			return <></>
+		case 'pending':
+			return <InfoPending />
+		case 'rejected':
+			return <InfoError displayText='Failed to load information' message={receipt.value.error.message} />
+		case 'resolved':
+			const transactionFee = calculateGasFee(receipt.value.value.effectiveGasPrice, receipt.value.value.gasUsed)
+			return (
+				<>
+					<Info label='Transaction Fee' value={`${transactionFee} ETH`} />
+				</>
+			)
+	}
+}
+
+const AccountBalance = ({ response }: { response: TransactionResponse }) => {
+	const providers = useProviders()
+	const { value: asyncBalance, waitFor } = useAsyncState<BigNumber>()
+
+	const getBalance = () => {
+		const { from, blockNumber } = response
+		waitFor(async () => {
+			const provider = providers.getbrowserProvider()
+			return await provider.getBalance(from, blockNumber)
+		})
+	}
+
+	useEffect(() => {
+		getBalance()
+	}, [response.blockNumber])
+
+	switch (asyncBalance.value.state) {
+		case 'inactive':
+			return <></>
+		case 'pending':
+			return <InfoPending />
+		case 'rejected':
+			return <InfoError displayText='Failed to load information' message={asyncBalance.value.error.message} />
+		case 'resolved':
+			const balance = ethers.utils.formatEther(asyncBalance.value.value)
+			return <Info label='Balance Before Transaction' value={balance} suffix=' ETH' />
+	}
+}
+
+function useTransaction(transactionHash: string) {
+	const providers = useProviders()
+	const { value: transactionResponse, waitFor: waitForResponse, reset: resetResponse } = useAsyncState<TransactionResponse>()
+	const { value: transactionReceipt, waitFor: waitForReceipt, reset: resetReceipt } = useAsyncState<TransactionReceipt>()
+
+	const getTransactionResponse = (transactionHash: string) => {
+		waitForResponse(async () => {
+			const provider = providers.getbrowserProvider()
+			const result = await provider.getTransaction(transactionHash)
+			// TransactionResult can actually be null?
+			if (result === null) throw new Error('Transaction was not found in chain!')
+			return result
+		})
+	}
+
+	const getTransactionReceipt = (txResponse: TransactionResponse) => {
+		waitForReceipt(async () => {
+			return await txResponse.wait()
+		})
+	}
+
+	// automatically get transaction receipt
+	useSignalEffect(() => {
+		if (transactionResponse.value.state !== 'resolved') return
+		getTransactionReceipt(transactionResponse.value.value)
+	})
+
+	// reset async states
+	useEffect(() => {
+		resetResponse()
```

Comment:
This is unnecessary since you call `gettransactionResponse` immediately below.  `useAsyncState` handles all of the issues around firing off multiple requests in a row before receiving a response (it drops the old ones and uses the new one).